src/components/Course/Placeholder.vue

```diff
@@ -15,17 +26,48 @@
 <script lang="ts">
 import { PropType, defineComponent } from 'vue';
 import CourseCaution from '@/components/Course/CourseCaution.vue';
+import NewSelfCheckOrPlaceholderCourseModal from '@/components/Modals/NewCourse/NewSelfCheckOrPlaceholderCourseModal.vue';
 
 export default defineComponent({
-  components: { CourseCaution },
+  components: { CourseCaution, NewSelfCheckOrPlaceholderCourseModal },
   props: {
     placeholderObj: { type: Object as PropType<FirestoreSemesterPlaceholder>, required: true },
     compact: { type: Boolean, required: true },
     semesterIndex: { type: Number, required: false, default: 0 },
   },
+  emits: {
+    'delete-placeholder': (placeholderObj: FirestoreSemesterPlaceholder) =>
+      typeof placeholderObj === 'object',
+  },
+  data() {
+    return {
+      isPlaceholderModalOpen: false,
+    };
+  },
   methods: {
     openModal() {
-      // TODO: open modal to assign a course to a placeholder
+      this.isPlaceholderModalOpen = true;
+    },
+    closeModal() {
+      this.isPlaceholderModalOpen = false;
+    },
+    assignCourse() {
+      // TODO @willespencer write the code that assigns a course to a placeholder
```

Comment:
How were we going to implement assigning courses to placeholders? I thought we modeled placeholders after normal courses because we were going to replace them -- I don't know if this added complexity is worth it.

Reply:
Sorry - assign might be the wrong choice here! Yes the idea was to insert and replace a placeholder with a course in the courses list.